spec/runner_spec.rb

```diff
@@ -16,11 +16,37 @@ def on_message(&block)
   def process(message)
     @messages << message
 
-    processor = @processor_queue.shift || proc {} 
+    processor = @processor_queue.shift || proc {}
     processor.call(message)
   end
 end
 
+class TestBatchConsumer < Racecar::Consumer
+  attr_reader :messages
+
+  def initialize
+    @messages = []
+    @processor_queue = []
+  end
+
+  def on_message(&block)
+    @processor_queue << block
+    self
+  end
+
+  def process_batch(messages)
+    @messages += messages
```

Comment:
It would be a good idea to add an example batch consumer to `examples/` – you can run it with

    ruby -Ilib exe/racecar --require examples/batch_consumer BatchConsumer